spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -168,6 +174,17 @@ public StructType schema() {
     return CAPABILITIES;
   }
 
+  @Override
+  public MetadataColumn[] metadataColumns() {
+    DataType sparkPartitionType = SparkSchemaUtil.convert(Partitioning.partitionType(table()));
+    return new MetadataColumn[] {
+        new SparkMetadataColumn(MetadataColumns.SPEC_ID.name(), DataTypes.IntegerType, false),
+        new SparkMetadataColumn(MetadataColumns.PARTITION_COLUMN_NAME, sparkPartitionType, true),
+        new SparkMetadataColumn(MetadataColumns.FILE_PATH.name(), DataTypes.StringType, false),
+        new SparkMetadataColumn(MetadataColumns.ROW_POSITION.name(), DataTypes.LongType, false)
```

Comment:
I am not too familiar with the metadata columns, but I see 5 currently defined in Iceberg's `MetadataColumns`. Is there a reason to omit `_deleted` here? And it probably doesn't matter, but should we keep the columns in the order defined in `MetadataColumns` (ordered by id from -1 to -5)?

Reply:
I can match the order in `MetadataColumns` for consistency.

I am not sure how useful `_deleted` metadata column will be in Spark now. I guess it will be always `false`?
@jackye1995 @chenjunjiedada @RussellSpitzer @rdblue, any thoughts?

Reply:
`_deleted` can be added later. The purpose of that field is to allow us to merge deletes in actions.